clients/src/main/java/org/apache/kafka/common/ClusterResourceListeners.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The
+ * ASF
+ * licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in
+ * compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License
+ * is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ClusterResourceListeners {
+
+    private final List<ClusterResourceListener> clusterResourceListeners;
+
+    public ClusterResourceListeners() {
+        this.clusterResourceListeners = new ArrayList<>();
+    }
+
+    public List<ClusterResourceListener> getClusterResourceListeners() {
+        return clusterResourceListeners;
+    }
+
+    public void add(Object candidate) {
```

Comment:
It is because this method needs to accept all types (MetricReporter, Interceptors, Serializers etc) that might or might not implement the ClusterListenerInterface.